crates/edr_evm/src/mempool.rs

```diff
@@ -335,6 +335,10 @@ impl MemPool {
                     let mut invalidated_transactions = pending_transactions.split_off(idx + 1);
                     let removed = pending_transactions.remove(idx);
 
+                    if pending_transactions.is_empty() {
+                        self.pending_transactions.remove(caller);
+                    }
```

Comment:
I'm curious why this was needed. I would've guessed that having an entry in the pending/future transactions map whose value is an empty list wouldn't be a problem.